### PR TITLE
Glass Flooring

### PR DESCRIPTION
**Pull request by Andrew-Fall**
Creates glass floors, which are transparent and display the Z level below them, while blocking air and supporting entities.

### CODE REVIEW

**Comment by Andrew-Fall** [comment]
I need to make deconstruction take time, as smacking it with a wrench can decompress a location.

File: code/controllers/subsystems/zcopy.dm [hunk 1]
```diff
@@ -135,14 +135,20 @@ SUBSYSTEM_DEF(zcopy)
...
 			Td = Td.below
+			if(!(Td.z_flags & ZM_MIMIC_OVERWRITE))
+				overwrite_override = TRUE
 
 		T.z_depth = depth = min(T.z - Td.z, OPENTURF_MAX_DEPTH)
 
 		var/t_target = OPENTURF_MAX_PLANE - depth	// this is where the openturf gets put
 
+		if(overwrite_override)
+			t_target = OPENTURF_MAX_PLANE - 1
```
Comment (Lohikar) [inline code/controllers/subsystems/zcopy.dm:150]: What is this for?

Comment (Andrew-Fall) [reply code/controllers/subsystems/zcopy.dm:150]: I forgot I made this hack cause zcopy isn't doing the planes in the correct order or something

Comment (Andrew-Fall) [reply code/controllers/subsystems/zcopy.dm:150]: Underlying zcopy issues are currently being addressed, holding off on this until that's done.

Comment (MarinaGryphon) [reply code/controllers/subsystems/zcopy.dm:150]: I, too, had problems with layering when I tried to port glass floors. Still unsure what happened and why.

**Comment by Andrew-Fall** [comment]
Closing for now

**Pull request closed without merging**